service/resource/namespace/resource.go

```diff
@@ -93,8 +93,12 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 	// further work. Then we cancel the reconciliation to prevent the current and
 	// any further resource from being processed.
 	if namespace.Status.Phase == "Terminating" {
+		r.logger.Log("cluster", key.ClusterID(customObject), "debug", "namespace is in state 'Terminating'")
+
 		canceler, exists := cancelercontext.FromContext(ctx)
 		if exists {
+			r.logger.Log("cluster", key.ClusterID(customObject), "debug", "canceling reconciliation for custom object")
+
 			canceler <- struct{}{}
```

Comment:
why not `close(canceler)`?

Reply:
Unrelated to this PR. This PR is about logging. There was no special reason. I just went with this. Thinking about it I think channels should only be closed where they are created, but I might be wrong.

Reply:
AFAIK you shouldn't close channels that you receive from. But for that you have receive only channels. When you can communicate something, you can also communicate that you have nothing more to communicate which is closing.

Reply:
Good point. We could then also go with just closing since the channel has to be created for each custom object reconciliation anyway.